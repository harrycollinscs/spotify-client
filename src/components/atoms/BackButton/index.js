import React from "react";
import { faArrowLeft } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import styled from "styled-components";

const IconContainer = styled.div`
  height: 35px;
  width: 35px;
  border-radius: 45px;
  background-color: rgb(255, 255, 255, 0.5);
  align-items: center;
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
`;

const IconStyle = {
  color: "#fff",
  margin: "5px 5px 5px 5px",
  height: "20px",
  width: "20px",
};

const goBack = () => {
  window.history.back();
};

const BackButton = props => (
  <IconContainer onClick={goBack}>
    <FontAwesomeIcon icon={faArrowLeft} style={IconStyle} />
  </IconContainer>
);

export default BackButton;
