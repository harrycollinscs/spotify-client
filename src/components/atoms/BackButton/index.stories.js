import React from "react";
import BackButton from "./index";

export default {
  title: "Atoms/BackButton",
  component: BackButton,
};

const Template = () => <BackButton />;

export const Default = Template.bind({});
