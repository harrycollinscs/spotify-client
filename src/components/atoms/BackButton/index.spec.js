import React from 'react';
import BackButton from './index';
import { shallow } from 'enzyme';

describe("Atoms/BackButton", () => {
  it("Should match snapshot", () => {
    const render = shallow(<BackButton />);
    expect(render).toMatchSnapshot();
  });
});
