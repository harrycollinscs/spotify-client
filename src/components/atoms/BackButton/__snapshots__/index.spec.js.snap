// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/BackButton Should match snapshot 1`] = `
<styled.div
  onClick={[Function]}
>
  <FontAwesomeIcon
    border={false}
    className=""
    fixedWidth={false}
    flip={null}
    icon={
      Object {
        "icon": Array [
          448,
          512,
          Array [],
          "f060",
          "M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z",
        ],
        "iconName": "arrow-left",
        "prefix": "fas",
      }
    }
    inverse={false}
    listItem={false}
    mask={null}
    pull={null}
    pulse={false}
    rotation={null}
    size={null}
    spin={false}
    style={
      Object {
        "color": "#fff",
        "height": "20px",
        "margin": "5px 5px 5px 5px",
        "width": "20px",
      }
    }
    swapOpacity={false}
    symbol={false}
    title=""
    transform={null}
  />
</styled.div>
`;
